Framework/DataHandling/src/LoadNexusProcessed.cpp

```diff
@@ -1062,14 +1063,17 @@ API::Workspace_sptr LoadNexusProcessed::loadPeaksEntry(NXEntry &entry) {
     qSign = -1.0;
 
   for (int r = 0; r < numberPeaks; r++) {
-    Kernel::V3D v3d;
-    if (convention == "Crystallography")
-      v3d[2] = -1.0;
-    else
-      v3d[2] = 1.0;
-    Geometry::IPeak *p;
-    p = peakWS->createPeak(v3d);
-    peakWS->addPeak(*p);
+    // Warning! Do not use anything other than the default constructor here
+    // It is currently important (10/05/17) that the DetID (set in the loop
+    // below this one) is set before QLabFrame as this causes Peak to ray trace
+    // to find the location of the detector, which significantly increases
+    // loading times.
+    Geometry::IPeak *peak = new Peak();
+    const auto goniometer = peakWS->run().getGoniometer();
+    peak->setInstrument(peakWS->getInstrument());
+    peak->setGoniometerMatrix(goniometer.getR());
+    peak->setRunNumber(peakWS->getRunNumber());
+    peakWS->addPeak(*peak);
```

Comment:
Should we add something like `void PeaksWorkspace::addPeak(Geometry::Peak &&peak)` to move, rather than copy, each `Peak`?